### PR TITLE
feat(google-calendar): add logic for multiple vacation days

### PR DESCRIPTION
**Pull request by PatrickDohn**
#### Summary

_Enumerate and describe your changes_

- Add ability to set vacation days in the future

#### Checklist

_Double check list and complete these actions before merge_

- [x] PR title is in format `TYPE(scope): description`
- [x] Link PR to Issue (if any issues are related)

### CODE REVIEW

File: src/app.js [hunk 1]
```diff
@@ -249,12 +257,40 @@ app.action(SELECT_PAYMO_PROJECT_ID, async (response) => {
   }
 });
 
+app.action('start-date', async (response) => {
```
Comment (davidholyko) [inline src/app.js:260]: we dont want strings like this written out. theyre prone to typos and other bugs.

use a variable that represents the string. Look at the constants file for reference

File: src/app.js [hunk 2]
```diff
@@ -249,12 +257,40 @@ app.action(SELECT_PAYMO_PROJECT_ID, async (response) => {
...
+  await response.ack();
+});
+app.action('end-date', async (response) => {
```
Comment (davidholyko) [inline src/app.js:263]: since these two have same no-op action handler, they can listen to the same event

File: src/app.js [hunk 3]
```diff
@@ -249,12 +257,40 @@ app.action(SELECT_PAYMO_PROJECT_ID, async (response) => {
...
+  await response.ack();
+});
+
+app.action('date-submit', async (response) => {
```
Comment (davidholyko) [inline src/app.js:267]: ditto use a variable that represents the string

File: src/app.js [hunk 4]
```diff
@@ -249,12 +257,40 @@ app.action(SELECT_PAYMO_PROJECT_ID, async (response) => {
...
+  const value = response.body.state.values;
+  const key = Object.keys(value);
+  const { body, client } = response;
```
Comment (davidholyko) [inline src/app.js:270]: nit move restructuring lines above

File: src/app.js [hunk 5]
```diff
@@ -249,12 +257,40 @@ app.action(SELECT_PAYMO_PROJECT_ID, async (response) => {
...
+  const { chat } = client;
+  const channel = body.user.id;
+  const startDate = response.body.state.values[key]['start-date']['selected_date'];
+  const endDate = response.body.state.values[key]['end-date']['selected_date'];
+  await response.ack();
```
Comment (davidholyko) [inline src/app.js:275]: `await response.ack();` should  be the first liine in this function

File: src/app.js [hunk 6]
```diff
@@ -249,12 +257,40 @@ app.action(SELECT_PAYMO_PROJECT_ID, async (response) => {
...
+  try {
```
Comment (davidholyko) [inline src/app.js:276]: try should encapsulate lines above

File: src/utils/googleCalendarUtil.js [hunk 7]
```diff
@@ -89,6 +89,36 @@ class GoogleCalendarUtil {
       },
     );
   }
+
+  async createFutureVacationEvent(slackName, startDate, endDate) {
+    console.log('createFutureVacationEvent', startDate, endDate);
```
Comment (davidholyko) [inline src/utils/googleCalendarUtil.js:94]: rm

File: src/utils/googleCalendarUtil.js [hunk 8]
```diff
@@ -89,6 +89,36 @@ class GoogleCalendarUtil {
...
+    const event = {
```
Comment (davidholyko) [inline src/utils/googleCalendarUtil.js:95]: this function looks familiar, i think we can make the single day function more generic and then have this use that

File: src/utils/googleCalendarUtil.js (hunk 7 above)
Comment (PatrickDohn) [reply src/utils/googleCalendarUtil.js:94]: fixed

File: src/app.js (hunk 1 above)
Comment (PatrickDohn) [reply src/app.js:260]: Fixed

File: src/utils/slackMessageUtil.js [hunk 9]
```diff
@@ -306,6 +310,53 @@ class SlackMessageUtil {
...
+        type: 'actions',
+        elements: [
+          {
+            type: 'button',
+            text: {
+              type: 'plain_text',
+              text: 'Submit',
+              emoji: true,
+            },
+            value: 'click_me_123', // TODO: figure out what value should be
```
Comment (davidholyko) [inline src/utils/slackMessageUtil.js:352]: possibly do this in separate ticket

File: src/utils/googleCalendarUtil.js (hunk 8 above)
Comment (davidholyko) [reply src/utils/googleCalendarUtil.js:94]: possibly do this in separate ticket

**Review by davidholyko — APPROVED** [review]
LGTM

**Pull request merged**